kani-compiler/src/main.rs

```diff
@@ -104,10 +107,19 @@ fn main() -> Result<(), &'static str> {
     );
 
     // Generate rustc args.
-    let rustc_args = generate_rustc_args(&matches);
+    let mut rustc_args = generate_rustc_args(&matches);
 
-    if matches.is_present(parser::ENABLE_STUBBING) {
-        eprintln!("warning: Kani currently does not perform any stubbing.");
+    // If appropriate, collect and set the stub mapping.
+    if matches.is_present(parser::ENABLE_STUBBING)
+        && queries.get_reachability_analysis() == ReachabilityType::Harnesses
```

Comment:
Sounds good!

Reply:
Actually, I'm not sure if we want to require `ReachabilityType::Harnesses` for `--enable-stubbing`. This will fail on a test I have involving `cargo kani` running on multiple crates (one of which is running with `ReachabilityType::None`).

Reply:
Oh.. where is the code for calling this via cargo? Are you adding `--enable-stubbing` as part of `pkg_args`?

Reply:
I don't think I've done anything to `pkg_args`. The test that invokes everything using cargo is `tests/cargo-kani/foreign-function-stubbing/`. The arguments (including `--enable-stubbing`) are set in the `Cargo.toml` there.

Reply:
Sorry if I wasn't clear. The `--enable-stubbing` in that file will be passed to `kani-driver` only. I'm trying to understand how this is betting passed down to `kani-compiler`. There are basically two ways today to pass information to the compiler, one as part of `KANI_FLAGS` and a second one via `cargo rustc -- <extra_flags>`. I can provide more context later if you want, or you can find more details in the MIR Linker RFC.

That said, I think passing the `--enable-stubbing` as part of `pkg_args` will probably be more painful today, so let's just leave it as is for now.

Reply:
Oh okay, sorry for the confusion: `--enable-stubbing` is passed through `KANI_FLAGS`.